astropy/utils/exceptions.py

```diff
@@ -42,3 +42,16 @@ class AstropyBackwardsIncompatibleChangeWarning(AstropyWarning):
     The suggested procedure is to issue this warning for the version in
     which the change occurs, and remove it for all following versions.
     """
+
+class _NoValue:
```

Comment:
So, is this going to be removed in the future when Numpy has fixed its `repr`? If so, a note should be added. Or if not, maybe a note should be added too?

Reply:
Nope, this is not independent of Numpy (basically we've just copied their idea as we shouldn't really use the non-public np._NoValue...), but this can be updated depending if they cook up a nicer way to workaround a nice way to do repr.

Reply:
No this is our `_NoValue` now :) At least if I understand it correctly.

Reply:
Yes, that's right. We can remove it when no args are being deprecated any more (and we decide never do a deprecation again in the future).

Reply:
Never say never...

Reply:
Well, it would also provide a nice way to handle more keyword-only arguments (ref: #6682). I had to skip some arguments because `None` and not-given would behave differently.